src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3046,6 +3046,15 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
                 {
                     var existingOrAddedMembers = addProperties(ctor.Parameters);
                     addDeconstruct(ctor, existingOrAddedMembers);
+
+                    foreach (var parameter in ctor.Parameters)
+                    {
+                        var parameterType = parameter.Type;
+                        if (parameterType.IsPointerOrFunctionPointer() || parameterType.IsRestrictedType())
```

Comment:
>IsPointerOrFunctionPointer [](start = 42, length = 26)

Should we use ```IsUnsafe``` helper instead? It looks like this is what ```GetAnonymousTypeFieldType``` is using and it looks like ```IsUnsafe``` flags more types.

Reply:
The question is should we align with type argument rules or anonymous type rules. 
I included tests with `int*[]` to highlight the difference.

Raised question in thread with LDM

---
In reply to: [512784802](https://github.com/dotnet/roslyn/pull/48925#discussion_r512784802) [](ancestors = 512784802)

Reply:
Thread from LDM landed on disallowing *unsafe* types. Pushed the change to the PR. Thanks

---
In reply to: [513087857](https://github.com/dotnet/roslyn/pull/48925#discussion_r513087857) [](ancestors = 513087857,512784802)